parachain/pallets/commitments/src/lib.rs

```diff
@@ -1,42 +1,41 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use sp_std::prelude::*;
+use artemis_core::{BasicMessageCommitment, ChannelId, IncentivizedMessageCommitment};
+use codec::{Decode, Encode};
+use ethabi::{self, Token};
 use frame_support::{
-	decl_module, decl_storage, decl_event, decl_error,
-	weights::Weight,
-	dispatch::DispatchResult,
+	decl_error, decl_event, decl_module, decl_storage, dispatch::DispatchResult, weights::Weight,
 };
+
+use sp_core::{RuntimeDebug, H160, H256};
 use sp_io::offchain_index;
-use sp_core::{H160, H256, RuntimeDebug};
 use sp_runtime::{
 	traits::{Hash, Zero},
-	DigestItem
+	DigestItem,
 };
-use codec::{Encode, Decode};
-use artemis_core::{ChannelId, MessageCommitment};
-use ethabi::{self, Token};
-use enum_iterator::IntoEnumIterator;
+use sp_std::prelude::*;
 
+use merkle_tree::*;
+
+mod merkle_tree;
 #[cfg(test)]
 mod mock;
-
 #[cfg(test)]
 mod tests;
 
 /// Auxiliary [`DigestItem`] to include in header digest.
 #[derive(Encode, Decode, Copy, Clone, PartialEq, RuntimeDebug)]
 pub enum AuxiliaryDigestItem {
```

Comment:
Lets move `AuxiliaryDigestItem` to `primitives/core` so it can be reused by the both outbound channels which need to generate their own commitments.